Machine Learning for Audio Classification (#3634)
section-engineering-education/engineering-education: content/articles/machine-learning-for-audio-classification/index.md

In the code below, we repeat the steps we used earlier to preprocess audio data. We then perform a prediction of the class it belongs to and finally uses the `inverse_transform` method from scikitlearn to give us the name of the predicted label. 

```python
filename="fold8/103076-3-0-0.wav"
audio, sample_rate = librosa.load(filename, res_type='kaiser_fast') 
mfccs_features = librosa.feature.mfcc(y=audio, sr=sample_rate, n_mfcc=40)
mfccs_scaled_features = np.mean(mfccs_features.T,axis=0)

print(mfccs_scaled_features)
mfccs_scaled_features=mfccs_scaled_features.reshape(1,-1)
print(mfccs_scaled_features)
print(mfccs_scaled_features.shape)
predicted_label=model.predict(mfccs_scaled_features)
print(predicted_label)
classes_x=np.argmax(predicted_label,axis=1)
prediction_class = labelencoder.inverse_transform(classes_x)
prediction_class
```
```bash
[[1.1630526e-21 5.3596851e-09 2.6831966e-09 9.9984801e-01 4.6294679e-09
  9.3139047e-12 1.5179862e-04 4.0151480e-34 1.1097348e-07 4.4551591e-08]]

array(['dog_bark'], dtype='<U16')
```
The model has correctly predicted the dog bark.

### Wrapping up

> Audio signal processing is not easy to understand as you need to have at least some domain knowledge. But it isn't difficult either. Using Python and libraries such as librosa, which have done most of the hard work for you, becomes easy to understand. You don't have to use the librosa library for this task. Once you have the waveform, you could convert the waveform into a spectrogram and use a Convolution Neural Network (CNN) for classification instead.

Reviewer: ```suggestion
Audio signal processing is not easy to understand as some domain knowledge is required. However, using libraries such as librosa, it becomes easy to understand. You don't have to use the librosa library for this task. Once you have the waveform, you could convert it into a spectrogram and use a Convolution Neural Network (CNN) for classification instead.
```
